LUCENE-10098: add note/link to GermanAnalyzer for decompounding nouns.

rmuir opened the pull request:
We can't do this out of box with the analyzer, due to incompatible
licenses. But we can make it easy on the user to do this, by linking to
repo that has sample code, documentation, and the required data files.

rmuir committed:
LUCENE-10098: add note/link to GermanAnalyzer for decompounding nouns.

We can't do this out of box with the analyzer, due to incompatible
licenses. But we can make it easy on the user to do this, by linking to
repo that has sample code, documentation, and the required data files.

uschindler commented:
I think this is a good addition.

As owner of the linked repository, I tried to get some more details on the licenses, but it is impossible to include both files needed into Lucene. The dictionary is derived work (done by myself), but the original dictionary is LGPL.

Maybe we should add another sentence to explain that the decompounding requires data files which are incompatible to ASF requirements.

uschindler commented:
+1 to Backport (if still possible for 8.10).

rmuir committed:
LUCENE-10098: move to 8.11 and add note about data file license

rmuir committed:
LUCENE-10098: fix artifact of 'gradlew tidy'

rmuir commented:
Thanks @uschindler . I added a phrase about the data files having licenses incompatible with Apache 2.0. I also moved to 8.11 (copying the 8.11 section from `branch_8x`, it was missing in `main`).
